graphql/pupil/mutations.ts

```diff
@@ -361,4 +363,43 @@ export class MutatePupilResolver {
         await invalidatePupilScreening(pupilScreeningId);
         return true;
     }
+
+    @Mutation(() => Boolean)
+    @Authorized(Role.ADMIN)
+    async adminIncreasePupilGrades() {
+        const pupils = await prisma.pupil.findMany({
+            where: {
+                grade: { not: null, notIn: ['13. Klasse', '14. Klasse'] },
+                OR: [{ gradeUpdatedAt: { lt: moment().subtract(10, 'months').toDate() } }, { gradeUpdatedAt: { equals: null } }],
+            },
+        });
+        logger.info(`Attempting to increase the grade of ${pupils.length} pupils`);
+        let validCount = 0;
+        const batchSize = 1000;
+        for (let i = 0; i < pupils.length; i += batchSize) {
+            const batch = pupils.slice(i, i + batchSize);
+
+            await Promise.all(
+                batch.map(async (pupil) => {
+                    try {
+                        const grade = gradeAsInt(pupil.grade);
+                        await prisma.pupil.update({
+                            where: { id: pupil.id },
+                            data: {
+                                grade: `${grade + 1}. Klasse`,
+                                gradeUpdatedAt: new Date(),
```

Comment:
I decided to set only this value here since it seems we want to update everyone's grade by 1 no matter if the user recently updated it... which is a bit weird.. But in general, all this grade stuff is a bit weird.

>Increment grade on August 1 by 1 - no matter when the grade was last updated

https://www.notion.so/lern-fair/Automatische-Anpassung-der-Jahrgangsstufe-ee4c95a8b8564cb08841aa8ec6d319d9?pvs=4#b1c9f041e09846bf81528742238075fa

Reply:
I also find it weird and also would prefer to respect it if a user updated it recently - e.g. if they register during the summer break and already take their "new class". Maybe double check with Fredrik?